src/normal.c

```diff
@@ -2562,7 +2562,14 @@ nv_z_get_count(cmdarg_T *cap, int *nchar_arg)
 	if (nchar == K_DEL || nchar == K_KDEL)
 	    n /= 10;
 	else if (VIM_ISDIGIT(nchar))
+	{
+	    if (n > LONG_MAX / 10 - (nchar - '0'))
+	    {
+		clearopbeep(cap->oap);
+		break;
+	    }
 	    n = n * 10 + (nchar - '0');
+	}
```

Comment:
I think you should go back to the original: `if (n > LONG_MAX/10)`,
otherwise "9...9" is considered overflow one loop too early.

```
Precondition:   n * 10 + 9  <= max
    ==>         n <= (max - 9) / 10 
Invert!:        n > (max - 9) / 10
```
With some handwaving, I'd say you can drop the "- 9" because of the "/ 10".
So you end up with the original
```
    ==>         n > max / 10
```
The test I'm running gives the same result with both
`n > (max - 9) / 10` and `n > max / 10`.

BTW, don't want a variable in the comparison, only constants,
so it's simply a compare against a constant.

With "n > max / 10 - 9" it's wrong when the value is like '999...'

(I wrote a simple program, nested loops to try all kinds of values,
to discover and bolster an intuitive sense to reveal why/how I screwed up.
(And I'm still working on the intuitive sense))

In the following the number in parens is the last "valid" number befor aborting
due to overflow. Notice 
```
=== n > max / 10 - 9
 ...
 6 - OVERFLOW 1 - 688 (688) 8
 7 - OVERFLOW 1 - 788 (788) 8
 8 - OVERFLOW 1 - 888 (888) 8
 9 - OVERFLOW 1 - 98 (98) 8
 ...
 6 - OVERFLOW 1 - 699 (699) 9
 7 - OVERFLOW 1 - 799 (799) 9
 8 - OVERFLOW 1 - 899 (899) 9
 9 - OVERFLOW 1 - 99 (99) 9

=== n > max / 10 
 ...
 6 - OVERFLOW 1 - 688 (688) 8
 7 - OVERFLOW 1 - 788 (788) 8
 8 - OVERFLOW 1 - 888 (888) 8
 9 - OVERFLOW 1 - 988 (988) 8
 ...
 6 - OVERFLOW 1 - 699 (699) 9
 7 - OVERFLOW 1 - 799 (799) 9
 8 - OVERFLOW 1 - 899 (899) 9
 9 - OVERFLOW 1 - 999 (999) 9
```

Reply:
okay thanks done